src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -87,7 +87,7 @@ public class RecordService {
           });
 
   public static final ThreadLocal<SimpleDateFormat> TIME_FORMAT =
-      ThreadLocal.withInitial(() -> new SimpleDateFormat("HH:mm:ss.SSSZ"));
+      ThreadLocal.withInitial(() -> new SimpleDateFormat("HH:mm:ss.SSSXXX"));
```

Comment:
This needs to be updated for Snowpipe Streaming but I don't see a good way to distinguish Snowpipe VS Snowpipe Streaming in RecordService, happy to hear any suggestions

Reply:
simplest way to do it would be in TopicPartitionChannel ctor. since TopicPartitionChannel is only used in Snowpipe Streaming, just set a private field in RecordService to true. this will not change anything in snowpipe. Something like

```
public void setEnableSchematization(final boolean enableSchematization) {
    this.enableSchematization = enableSchematization;
  }
```

Reply:
I thought about that, but `convertToJson` is a static function :)

Reply:
Basically we need to add a boolean to `convertToJson` and update all the references, I'm debating on whether this is really needed. The difference between Z and XXX is that there will be a colon with XXX for the timezone, for example, Z = `02:24:00.000+0000` and XXX = `02:24:00.000+00:00`. In fact, it will fail if you try to insert `02:24:00.000+0000` to a TIME column so this is technically a bug. I'm not sure why there is no complains from customer though.

Reply:
out of scope of this PR, but RecordService does have a lot of overlap between snowpipe and streaming, might be useful to refactor or split it into two files for simplicity

Reply:
> Basically we need to add a boolean to convertToJson and update all the references, I'm debating on whether this is really needed. The difference between Z and XXX is that there will be a colon with XXX in the timezone, for example, Z = 02:24:00.000+0000 and XXX = 02:24:00.000+00:00. In fact, it will fail if you try to insert 02:24:00.000+0000 to a TIME column so this is technically a bug. I'm not sure why there is no complains from customer, probably because not many people is using logical types.

Thanks for detailed explanation. 
I would be careful to make changes in snowpipe code. 
Most folks havent complained because we dont have schematization in snowpipe. they have base table and then more tables on top of that. This change will break their downstream pipeline right?

Reply:
You might have to take a giant step and add another arg in `convertToJson` :( 
@sfc-gh-rcheng 's suggestion is the only way forward.

Reply:
Added as an argument, PTAL